Add `registerWarnHandler()` tests

Turbo87 opened the pull request:
This PR adds the missing tests from #8

Turbo87 committed:
Add `registerWarnHandler()` tests

rwjblue commented:
> not ok 20 Chrome 63.0 - ember-debug: Ember.warn does not throw

This test is failing on 2.8 and 2.12

Turbo87 commented:
😭  ... but ... why?

rwjblue commented:
Finally back at a computer, pulling this down now....

Commit:
Replace legacy QUnit 1.x API with 2.x version.

After the general dependency updates done recently we are pulling in
QUnit 2.x which no longer supports `setup` and `teardown`.

Commit:
Add helper method for retrieving HANDLERS baseline.

In Ember 2.1 the hacky
`_____HANDLERS__DO__NOT__USE__SERIOUSLY__I_WILL_BE_MAD` property is not
present (because native support exists). This helper method abstracts
away the details around _how_ the baseline "handlers" are found (so they
can be properly reset).
